src/tools/pub/publishing.md

```diff
@@ -326,6 +326,41 @@ then you can ignore the following warning from `dart pub publish`:
    version 2.12.0-0, consider publishing the package as a pre-release
    instead."*
 
+
+## Rectracting a package version {#retract}
+
+If you wish to prevent new package consumers from adopting a recently
+published version of your package, you may **retract** that package version.   
+
+A retracted package version is not deleted, it will still show up in the version 
+list of the package on pub.dev, in the "Retracted versions" section. Also, the 
+detailed view of that package version will have a clear **RETRACTED** badge.
+
+A package version can only be retracted within 7 days of publication. 
+The retracted version can be restored again within 7 days of retraction.
+To retract or restore a package version, sign in to pub.dev using a Google Account
+that's an uploader or verified publisher admin for the package. Then use the 
+**Admin** tab of the individual package to select a version to retract or restore.  
+
+If you are a package consumer and depend on a version that later gets retracted,
+you can still use it as long as it is present in your `pubspec.lock` file. 
+If you want to depend on a specific version that is already retracted you need 
+to pin the version in the `dependency_overrides` section in the `pubspec.yaml` file. 
+In any other case the retracted package version will not be taken into account by 
+the version solver.
+
+### When does it make sense to retract a package version?
+
+Retracting a package causes churn and has a somewhat negativ impact on consumers. 
+Hence, before retracting a version you should consider whether it is in fact the 
+right course of action instead of just publishing a new version. For instance, 
+if your package version contains a minor bug then retraction is probably not 
+needed. On the other hand, if you accidentally published a major version or 
+published a version with wrong or missing dependency constraints, then retration 
+might be the right solution. 
+
+This feature is available from *Dart 2.15*.
```

Comment:
Maybe, we need to clarify that this means Dart prior to 2.15 has no notion of retracted package versions and will use them regardless of whether or not they are in the `pubspec.lock`.

Reply:
Just to verify, is this dependent on the user's Dart version or the pubspec's SDK constraint?

Reply:
The retracted feature depends on the users Dart SDK version.